### PR TITLE
Bump Aztec version to v1.5.9

### PR DESCRIPTION
**Pull request by derekblank**
This PR updates the version of Aztec to v1.5.9 to incorporate updates to the [AztecAndroid Placeholder API](https://github.com/wordpress-mobile/AztecEditor-Android/pull/990/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5).

Related PRs:
- https://github.com/WordPress/gutenberg/pull/41828
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/4973

To test:
- Verify editor works as normal.

## Regression Notes
1. Potential unintended areas of impact
Aztec Editor

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
No automated tests added as we have deprecated the Aztec Editor.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [ ] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by derekblank** [comment]
@geriux My understanding is that this PR will also need to bump the [gutenbergMobileVersion in build.gradle](https://github.com/wordpress-mobile/WordPress-Android/blob/44cda65a84099238fa32fb79403bb42eb7ef128d/build.gradle#L13), too. Currently it is pointing at the 1.78.1 release. Based on looking at [previous Aztec version bumps](https://github.com/wordpress-mobile/WordPress-Android/pull/15618/files), should this version number be arbitrarily updated to the next alpha version (i.e., 1.78.2-alpha1), or wait to be updated until the gutenberg-mobile PRs have been merged?

**Comment by derekblank** [comment]
@planarvoid I added basic testing instructions and regression notes regarding testing the Aztec changes -- let me know if anything else regarding the Placeholder API should be added for testing or regression review. 🙇

**Comment by geriux** [comment]
> Based on looking at [previous Aztec version bumps](https://github.com/wordpress-mobile/WordPress-Android/pull/15618/files), should this version number be arbitrarily updated to the next alpha version (i.e., 1.78.2-alpha1), or wait to be updated until the gutenberg-mobile PRs have been merged?

We should merge first the Gutenberg PR, then continue with the Gutenberg Mobile PR, create an alpha tag e.g. `v1.79.0-alpha1` and finally update this PR with it before merging. Before doing so we should check if we are doing a release anytime soon or not because once an alpha tag is merged, it gets complicated if we are skipping a release 😅 . If we skip a release we could create an _after_ branch to merge this PR.

**Comment by geriux** [comment]
It looks like the CI checks are failing because we need to point to the Guteberg mobile commit that has the Aztec update.

**Comment by derekblank** [comment]
> It looks like the CI checks are failing because we need to point to the Guteberg mobile commit that has the Aztec update.

@geriux Thanks -- I've created the [`1.79.0-alpha1`](https://github.com/wordpress-mobile/gutenberg-mobile/releases/tag/v1.79.0-alpha1) (which is pointing at the updated AztecAndroid v1.5.9), and updated the gutenberg ref in this PR. Let me know if you happen to note anything unusual with the BuildKite failures above. Clicking the "Details" link for each of the failures results in "Page Not Found" on BuildKite for me. Perhaps you can let me know if they need to be triggered to re-run some other way? If not, it may be an actual failure and I can look into that.

**Comment by geriux** [comment]
> Clicking the "Details" link for each of the failures results in "Page Not Found" on BuildKite for me. Perhaps you can let me know if they need to be triggered to re-run some other way? If not, it may be an actual failure and I can look into that.

So there was an error about a missing implementation, more speficially `beforeMediaDeleted` that was added in Aztec. I [updated the code](https://github.com/wordpress-mobile/WordPress-Android/pull/16780/commits/ec2a7a4272d611fcfced9d393da55a9bf3bea56e) to include it, let's see if buildkite success this time. If it does, would you mind giving it another test to check if all works ok? I did some testing and it's working fine but another would be good. Thank you!

**Comment by derekblank** [comment]
@geriux Yep, my testing was good too, and it looks like buildkite is succeeding. Thanks for all of the testing and review! I see that review approval is not required to merge, but I just wanted to double-check that that would be ok to self-merge before I do so. (I can also wait for approval of course, just curious what typically happens for WPAndroid.) 🙇

**Review by geriux — APPROVED** [review]
LGTM! Nice work!

**Comment by geriux** [comment]
> just curious what typically happens for WPAndroid.

We usually wait for an approval yes, just approved it! 🚀

**Pull request merged**
